airbyte-cdk/python/airbyte_cdk/connector_builder/migrator/source.py

```diff
@@ -0,0 +1,115 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import json
+import logging
+import os
+import re
+from enum import Enum
+from typing import Any, Dict, List, Optional, Tuple
+
+import ruamel.yaml
+from airbyte_cdk.sources.declarative.models.declarative_component_schema import AuthFlow, Spec
+
+logger = logging.getLogger("migrator.source")
+
+
+class SourceRepository:
+
+    def __init__(self, base_path: str) -> None:
+        self._base_path = base_path
+
+    def merge_spec_inside_manifest(self, source_name: str) -> None:
+        spec = self._fetch_spec_outside_manifest(source_name)
+        if spec:
+            manifest_path = self._manifest_path(source_name)
+            yaml = ruamel.yaml.YAML()
+            yaml.preserve_quotes = True
+            yaml.indent(mapping=2, sequence=4, offset=2)
```

Comment:
are these properly indenting the resulting manifests? the one that stuck out was sequence, but going off some of our existing manifests, I would think the mapping=1, sequence=2, offset=2?

Quoted by chatgpt:
mapping: Specifies the number of spaces used for each level of indentation in mappings (dictionaries).
sequence: Specifies the number of spaces used for each level of indentation in sequences (lists).
offset: Specifies the number of spaces used for indentation when a sequence is a value in a mapping.

But also sort of trust but verify, if it indents properly for you then chatgpt might just be wrong